OpenRA.Mods.OpenE2140/Warheads/SpawnWeaponsWarhead.cs

```diff
@@ -0,0 +1,158 @@
+﻿#region Copyright & License Information
+
+/*
+ * Copyright (c) The OpenE2140 Developers and Contributors
+ * This file is part of OpenE2140, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+
+#endregion
+
+namespace OpenRA.Mods.OpenE2140.Warheads;
+
+using Common.Warheads;
+using GameRules;
+using JetBrains.Annotations;
+using OpenRA.Mods.Common;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Mods.OpenE2140.Effects;
+using OpenRA.Primitives;
+using OpenRA.Traits;
+
+[UsedImplicitly(ImplicitUseTargetFlags.WithMembers)]
+[Desc("Spawn a weapon(s) on warhead impact.")]
+public class SpawnWeaponsWarhead : Warhead, IRulesetLoaded<WeaponInfo>
+{
+	[Desc("The weapons to spawn.")]
+	public readonly string[] Weapons = Array.Empty<string>();
+
+	[Desc("Delay in ticks before applying the warhead effect.", "0 = instant (old model).")]
+	public readonly int[] Delays = new int[] { 0 };
+
+	[Desc("The amount of projectile pieces to produce. Two values indicate a range.")]
+	public readonly int[] Pieces = { 3, 10 };
+
+	[Desc("The minimum and maximum distances the projectile may travel.")]
+	public readonly WDist[] Range = { WDist.FromCells(2), WDist.FromCells(5) };
+
+	[Desc("Whether to consider actors in determining whether the impact should happen. If false, only terrain will be considered.")]
+	public readonly bool ImpactActors = true;
+
+	internal WeaponInfo[] WeaponInfos { get; private set; } = Array.Empty<WeaponInfo>();
+
+	public void RulesetLoaded(Ruleset rules, WeaponInfo info)
+	{
+		if (this.Weapons.Length != this.Delays.Length)
+			throw new YamlException($"Length of: '{nameof(this.Weapons)}' and '{nameof(this.Delays)}' must be equal.");
+
+		this.WeaponInfos = this.Weapons.Select(w =>
+		{
+			var weaponToLower = w.ToLowerInvariant();
+			if (!rules.Weapons.TryGetValue(weaponToLower, out var weapon))
+				throw new YamlException($"Weapons Ruleset does not contain an entry '{weaponToLower}'");
+			return weapon;
+		}).ToArray();
+	}
+
+	public override void DoImpact(in Target target, WarheadArgs args)
+	{
+		if (target.Type == TargetType.Invalid)
+			return;
+
+		var firedBy = args.SourceActor;
+		var pos = new WPos(target.CenterPosition.X, target.CenterPosition.Y, Math.Max(0, target.CenterPosition.Z));
+		var world = firedBy.World;
+
+		var actorAtImpact = this.ImpactActors ? ActorTypeAtImpact(world, pos, firedBy) : ImpactActorType.None;
```

Comment:
```suggestion
		var actorAtImpact = this.ImpactActors ? this.ActorTypeAtImpact(world, pos, firedBy) : ImpactActorType.None;
```

Still missing `this`